aten/src/ATen/native/native_functions.yaml

```diff
@@ -7737,12 +7737,20 @@
 # TODO: remove dispatch section when porting TH CUDA to ATen
 - func: multinomial.out(Tensor self, int num_samples, bool replacement=False, *, Generator? generator=None, Tensor(a!) out) -> Tensor(a!)
   dispatch:
-    CPU, CUDA: multinomial_out
+    CPU, CUDA, MPS: multinomial_out
```

Comment:
It's much more convenient, because the code for multinomial without replacement is present in the same generic code, which MPS can also use (using other ops that are already implemented). If I registered multinomial_out_mps, I'd have to copy that code over, instead of the two-line is_mps change.